src/Factory/BaseModelFactory.php

```diff
@@ -50,10 +50,11 @@ public function create(array $data): Model
                 $this->formatType
             );
         } catch (Exception $invalidArgumentException) {
-            throw new InvalidArgumentException(
+            throw new InvalidDataException(
+                $data,
                 $invalidArgumentException->getMessage(),
                 $invalidArgumentException->getCode(),
-                $invalidArgumentException->getPrevious()
+                $invalidArgumentException
```

Comment:
Was this a bug? `getPrevious` returned `null` because there was no previous?

Reply:
yeah, I believe so, it should have been just the actual previous exception instead of the previous of the previous :D